candies/cli/cli.py

```diff
@@ -61,40 +61,87 @@ def help(self, *args, **kwargs):
         """
         return self.main.help(*args, **kwargs)
 
-    def parse(self, callable: Union[Callable[[str], Any], Type]) \
-            -> Union[Type, Callable[[str], Any]]:
-        """Defines a parser for a custom type.
+    def parse(self, func: Callable[[str], Any]) -> Callable[[str], Any]:
+        """Defines a parser function for a custom type.
+
+        This method is intended to be used as a function decorator for
+        defining parses for custom types (take a look at the `Examples`
+        section below).
+
+        Note:
+            If you want to add an existing function as a parser or
+            add a class with a constructor like `Some.__init__(str)`,
+            please refer to the `add_parser` method.
 
         Args:
-            callable: A callable that converts a string to an instance
-                of a custom type. For example, a parser function or a
-                type that accepts a string to a constructor.
+            func: A function that converts a string to an instance
+                of a custom type. Must have a return type annotation.
 
         Examples:
             @cli
             def add(money: Arg[Money]):
                 ...
 
-            @main.parse
+            @cli.parse
             def money(x: str) -> Money:
                 if x[0] == '$':
                     return Money(float(x[1:]), 'dollars')
                 ...
+        """
+        return self.add_parser(func)
+
+    def add_parser(self, callable: Union[Type, Callable[[str], Any]]) \
+            -> Union[Type, Callable[[str], Any]]:
+        """Adds a parser for a custom type.
+
+        When a function is provided, this method inspects it and considers
```

Comment:
Maybe the word `treats` would be better? `treats its return type as annotation ...`.

Reply:
Corrected!